commit 2ee32addb03e282035b5722c9216f5fb9e22d8f9
Author:     John Marshall <jmarshall@hey.com>
AuthorDate: Mon Sep 26 17:34:59 2022 +0100
Commit:     John Marshall <jmarshall@hey.com>
CommitDate: Mon Sep 26 17:34:59 2022 +0100

    Use plain mkstemp() if O_NOATIME is not available

    Platforms that don't have O_NOATIME (i.e., non-Linux) may not
    have mkostemp() either.

diff --git a/src/api.cpp b/src/api.cpp
index a66eb15..c44a5ad 100644
--- a/src/api.cpp
+++ b/src/api.cpp
@@ -17,11 +17,6 @@
 #define MMAP_FD_MASK 0x0fff
 #define MMAP_FLAG    0x1000
 
-/* O_NOATIME is defined at fcntl.h when supported */
-#ifndef O_NOATIME
-#define O_NOATIME 0
-#endif
-
 
 #define CHECK_FILE(filename, err) if(!is_file_exists(filename)) { \
                                       return err;                 \
@@ -198,8 +193,12 @@ void initialize_mat_full_no_biom_T(TMat* &result, const char* const * sample_ids
     } else {
       std::string mmap_template(mmap_dir);
       mmap_template+="/su_mmap_XXXXXX";
-      // note: mkostemp will update mmap_template in place
+      // note: mkstemp/mkostemp will update mmap_template in place
+#ifdef O_NOATIME
       int fd=mkostemp((char *) mmap_template.c_str(), O_NOATIME ); 
+#else
+      int fd=mkstemp((char *) mmap_template.c_str() );
+#endif
       if (fd<0) {
          result->matrix = NULL;
          // leave error handling to the caller
